pyteal/ast/subroutine.py

```diff
@@ -49,6 +49,16 @@ def __init__(
                     )
                 )
 
+        for var, var_type in implementation.__annotations__.items():
+            if var_type != Expr:
```

Comment:
```suggestion
            if var_type is not Expr:
```
Equality operators can "lie" by being overloaded, but `is` will never lie. Though I see you change this check in the follow up PR, so it's a bit of a moot point.